### DISCUSSION TITLE
mock library for testing

### DISCUSSION THREAD

**Discussion by 2bndy5** [adafruit/Adafruit_Blinka#877] [Show and tell]
I created a new library called [CircuitPython-mocks](https://circuitpython-mocks.rtfd.io/) which should allow collecting code coverage (and establishing expected code behavior) during soft-tests of Circuitpython-based projects.

Currently, the `SPI`, `I2C`, `UART`, and `DigitalInOut` classes are mocked (along with some dummy pins in a mocked `board` module). I'm interested in adding mocked PWM functionality, but first I was hoping to see if it is worth my time by gathering community feedback.

**Comment by dastels** [comment]
That's awesome!
